server/auth/session.guard.ts

```diff
@@ -42,24 +74,24 @@ export class SessionGuard extends BaseGuard {
                     this.logger.error(
                         `Affiliation mismatch: expected ${expectedAffiliation}, got ${appAffiliation}`
                     );
-                    try {
-                        const { data } = await ory.createBrowserLogoutFlow({
-                            cookie: request.header("Cookie"),
-                        });
-                        await ory.updateLogoutFlow({
-                            cookie: request.header("Cookie"),
-                            token: data.logout_token,
-                        });
-                    } catch (e) {
-                        this.logger.error("Error during logout flow", e);
-                    }
+                    await this.logoutUser(ory, request);
                     return this.checkAndRedirect(
                         request,
                         response,
                         isPublic,
                         "/unauthorized"
                     );
                 }
+
+                const mongoUserId = session?.identity?.traits?.user_id;
+                try {
+                    await this.usersService.getById(mongoUserId);
```

Comment:
suggestion: a better architecture is to always return something from a function, including errors, and treat them where you call it.